src/test/coinstatsindex_tests.cpp

```diff
@@ -79,6 +79,10 @@ BOOST_FIXTURE_TEST_CASE(coinstatsindex_initial_sync, TestChain100Setup)
     // Shutdown sequence (c.f. Shutdown() in init.cpp)
     coin_stats_index.Stop();
```

Comment:
Unrelated: I think the two `Stop()`, here and in fa176e253fb473767c61d4d8cd2d93e87d71a015 can be removed, as they are done by the destructor anyway.

Reply:
It looks like that all calls to `Stop()` are done just before the objects are destroyed (except the call from the destructor itself). Thus `Stop()` can be removed and its (short) body moved inside `BaseIndex::~BaseIndex()`. If this is the case then this deserves a separate PR. Am I missing something?

Reply:
The calls to Stop need to happen before the BaseIndex destructor runs because they are responsible for joining the sync thread, which can call virtual methods. By the time the BaseIndex destructor runs, whatever subclass inherited from BaseIndex is partially destroyed and the virtual method calls won't work.

The call to Stop() in the destructor was never valid and I have a commit c499c8797e65c66318eb1faa2e36c7c3f3e2ba1a in #24230 which removed it and asserts that it was called previously.